src/Components/Web.JS/src/Services/NavigationManager.ts

```diff
@@ -158,19 +158,23 @@ function navigateToCore(uri: string, options: NavigationOptions, skipLocationCha
 }
 
 function performExternalNavigation(uri: string, replace: boolean) {
-  if (location.href === uri) {
-    // If you're already on this URL, you can't append another copy of it to the history stack,
-    // so we can ignore the 'replace' flag. However, reloading the same URL you're already on
-    // requires special handling to avoid triggering browser-specific behavior issues.
-    // For details about what this fixes and why, see https://github.com/dotnet/aspnetcore/pull/10839
-    const temporaryUri = uri + '?';
-    history.replaceState(null, '', temporaryUri);
-    location.replace(uri);
-  } else if (replace) {
-    location.replace(uri);
-  } else {
-    location.href = uri;
-  }
+  // If we are navigating to an external URI, defer setting the URL so that we can
+  // complete the current navigation operation if the request comes from .NET.
+  setTimeout(() => {
```

Comment:
Is one downside to this approach (compared to the other one you proposed) that we don't actually wait for the "end .NET->JS interop" message to be sent across the wire before tearing down the page? i.e., there's still a possibility that the bug appears, especially when the user has an unreliable connection?

Reply:
We don't want to swallow the exception when the circuit just gets disconnected, do we?

Reply:
That's my reasoning for it. It's better to support the call finishing than to catch the exception on the server at the risk of catching exceptions that we don't want to catch. Specially since the outcome of navigation can't be observed by the user.